pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -152,6 +154,16 @@
     static <T> ConsumerImpl<T> newConsumerImpl(PulsarClientImpl client, String topic, ConsumerConfigurationData<T> conf,
                  ExecutorService listenerExecutor, int partitionIndex, CompletableFuture<Consumer<T>> subscribeFuture,
                  SubscriptionMode subscriptionMode, MessageId startMessageId, Schema<T> schema, ConsumerInterceptors<T> interceptors) {
+        if (schema != null && schema.supportSchemaVersioning()) {
```

Comment:
actually I am bit confused about the logic here. why do we need to cache this? 

Event we want to cache this, we should cache the `MultiVersionGenericSchemaProvider`. but we shouldn't cache the schema. because user can pass in different type of schema instances.